src/server/api/routers/rooms.ts

```diff
@@ -64,4 +64,23 @@ export const roomsRouter = createTRPCRouter({
       const [res] = await ctx.db.select().from(rooms).where(eq(rooms.id, input.id)).limit(1);
       return res;
     }),
+  delete: teacherProcedure
+    .input(
+      z.object({
+        id: z.string()
+      }),
+    )
+    .mutation( async ({ ctx, input }) => {
+      const checkClasses = await ctx.db.select().from(classes).where(eq(classes.roomId, input.id));
+
+      if (!checkClasses) {
+        const res = await ctx.db
+          .delete(rooms)
+          .where(eq(rooms.id, input.id))
+          .returning();
+        return res;
+      } else {
+        return ("You can't delete this room as there are classes scheduled.")
```

Comment:
Instead of returning a string, we need to throw an error here.
```suggestion
        throw new TRPCError({ code: "CONFLICT", message: "You can't delete this room as there are classes scheduled."})
```